### PR TITLE
check if the deb is in the pool in the job that releases the deb

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Checks if the deb is in the pool in the job that releases the deb.

### Motivation

> We currently have a step early in the gitlab pipeline that checks if the deb pkg is already in the pool, and fails if it's already in the pool (avoids hash sum mismatch errors).
> When there's a flake later in the pipeline though, pipelines that started later can push their deb package, and when the original is retried it won't check again for this case, and we'll hit the mismatch errors again.

### CODE REVIEW

File: .gitlab-ci.yml
Comment (olivielpeau) [inline .gitlab-ci.yml]: does the script work when the working dir isn't `$OMNIBUS_PACKAGE_DIR`?

**Review by olivielpeau — DISMISSED** [review]
Left a question, once addressed feel free to merge 👍 (as long as this actually works in a gitlab pipeline :) )

File: .gitlab-ci.yml
Comment (arbll) [reply .gitlab-ci.yml]: You're right, fixed that

**Comment by arbll** [comment]
Running a full pipeline would be quite hard so I simulated it: https://github.com/DataDog/datadog-agent/pull/2370/commits/0d006d0a22c76f15ba5cf40613d8ff04cbb5f182. Is that ok with you ?

Pipeline id is 7325343.

File: .gitlab-ci.yml
Comment (hush-hush) [inline .gitlab-ci.yml]: Why do we touch that file ?

Comment (arbll) [reply .gitlab-ci.yml]: https://github.com/DataDog/datadog-agent/commit/0d006d0a22c76f15ba5cf40613d8ff04cbb5f182 was added for testing. I used `touch` to create a fake `deb` that should make the pipeline fail.
This commit will be reverted before merging

Comment (hush-hush) [reply .gitlab-ci.yml]: :+1:

**Review by olivielpeau — DISMISSED** [review]
LGTM once you remove 0d006d0 :)

**Pull request merged**
